apple/swift-docc
# Add base support for converting catalogs that include multiple languages

Bug/issue #, if applicable: rdar://84169407

## Summary

Adds base support for converting DocC Catalogs that include multiple languages by expanding symbol semantic models to hold language-specific variants.

This builds on previous work landed in:
- #11
- #16
- #21 
- #25

## Dependencies

None.

## Testing

Build documentation for the included test bundle at `Tests/SwiftDocCTests/Test\ Bundles/MixedLanguageFramework.docc` and confirm that output is expected.

Steps:
1. export DOCC_HTML_DIR=/path/to/recent/[swift-docc-render-artifact](https://github.com/apple/swift-docc-render-artifact)
2. From the root of this repository:
     ```bash
     swift run docc preview Tests/SwiftDocCTests/Test\ Bundles/MixedLanguageFramework.docc \
         --enable-experimental-objective-c-support \
         --index
     ```
3. Confirm that mixed-language content renders as expected.

| Swift      | Objective-C |
| ----------- | ----------- |
| <img width="1464" alt="Screen Shot 2021-11-30 at 7 30 29 PM" src="https://user-images.githubusercontent.com/6750147/144167074-4969698c-e4b9-4d29-bb6e-e84f03e321ba.png">      |   <img width="1464" alt="Screen Shot 2021-11-30 at 7 30 36 PM" src="https://user-images.githubusercontent.com/6750147/144167101-0a83c432-e7a6-4eb4-93d8-52240a5c1349.png">    |





## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
